common/fixation_entry_test.go

```diff
@@ -0,0 +1,252 @@
+package common_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/store"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/common"
+	"github.com/lavanet/lava/common/types"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmdb "github.com/tendermint/tm-db"
+)
+
+// Helper function to init a mock keeper and context
+func initCtxAndFixationStore(t *testing.T) (*common.FixationStore, sdk.Context) {
+	db := tmdb.NewMemDB()
+	stateStore := store.NewCommitMultiStore(db)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+
+	mockStoreKey := sdk.NewKVStoreKey("storeKey")
+	mockMemStoreKey := storetypes.NewMemoryStoreKey("storeMemKey")
+	stateStore.MountStoreWithDB(mockStoreKey, sdk.StoreTypeIAVL, db)
+	stateStore.MountStoreWithDB(mockMemStoreKey, sdk.StoreTypeMemory, nil)
+
+	require.NoError(t, stateStore.LoadLatestVersion())
+
+	fixationKey := "mock_fix"
+
+	vs := common.NewFixationStore(mockStoreKey, cdc, fixationKey)
+
+	ctx := sdk.NewContext(stateStore, tmproto.Header{}, false, log.TestingLogger())
+
+	return vs, ctx
+}
+
+// Test addition and removal of a fixation entry
+func TestFixationEntryAdditionAndRemoval(t *testing.T) {
+	// create dummy data for dummy entry
+	dummyIndex := "index"
+	dummyObj := sdk.Coin{Denom: "utest", Amount: sdk.ZeroInt()}
+
+	// init FixationStore + context
+	vs, ctx := initCtxAndFixationStore(t)
+
+	// add dummy entry
+	blockToAddEntry := uint64(ctx.BlockHeight())
+	err := vs.AppendEntry(ctx, dummyIndex, blockToAddEntry, &dummyObj)
+	require.Nil(t, err)
+
+	// get all entry indices and make sure there is only one index
+	indexList := vs.GetAllEntryIndices(ctx)
+	require.Equal(t, 1, len(indexList))
+
+	// get the entry from the storage
+	var dummyCoin sdk.Coin
+	err = vs.GetEntry(ctx, dummyIndex, blockToAddEntry, &dummyCoin, types.DO_NOTHING)
+	require.Nil(t, err)
+
+	// make sure that one entry's data is the same data that was used to create it
+	require.True(t, dummyCoin.IsEqual(dummyObj))
+
+	// remove the entry by advancing over the STALE_ENTRY_TIME and appending a new one (append triggers the removal func)
+	ctx = ctx.WithBlockHeight(types.STALE_ENTRY_TIME + int64(blockToAddEntry) + 1)
```

Comment:
we need to also check that if we increase blocks but not as far as memory, we don't delete

Reply:
done